amaranth/back/rtlil.py

```diff
@@ -787,69 +787,71 @@ def _convert_fragment(builder, fragment, name_map, hierarchy):
+            "RD_CLK_ENABLE": _ast.Const(rd_clk_enable, max(1, len(fragment._read_ports))),
+            "RD_CLK_POLARITY": _ast.Const(rd_clk_polarity, max(1, len(fragment._read_ports))),
+            "RD_TRANSPARENCY_MASK": _ast.Const(rd_transparency_mask, max(1, len(fragment._read_ports) * len(fragment._write_ports))),
+            "RD_COLLISION_X_MASK": _ast.Const(0, max(1, len(fragment._read_ports) * len(fragment._write_ports))),
+            "RD_WIDE_CONTINUATION": _ast.Const(0, max(1, len(fragment._read_ports))),
+            "RD_CE_OVER_SRST": _ast.Const(0, max(1, len(fragment._read_ports))),
+            "RD_ARST_VALUE": _ast.Const(0, len(fragment._read_ports) * fragment._width),
+            "RD_SRST_VALUE": _ast.Const(0, len(fragment._read_ports) * fragment._width),
+            "RD_INIT_VALUE": _ast.Const(0, len(fragment._read_ports) * fragment._width),
+            "WR_PORTS": len(fragment._write_ports),
+            "WR_CLK_ENABLE": _ast.Const(wr_clk_enable, max(1, len(fragment._write_ports))),
+            "WR_CLK_POLARITY": _ast.Const(wr_clk_polarity, max(1, len(fragment._write_ports))),
+            "WR_PRIORITY_MASK": _ast.Const(0, max(1, len(fragment._write_ports) * len(fragment._write_ports))),
+            "WR_WIDE_CONTINUATION": _ast.Const(0, max(1, len(fragment._write_ports))),
         }
+        def make_en(port):
+            if len(port._data) == 0:
+                return _ast.Const(0, 0)
+            granularity = len(port._data) // len(port._en)
+            return _ast.Cat(en_bit.replicate(granularity) for en_bit in port._en)
         port_map = {
             "\\RD_CLK": _ast.Cat(rd_clk),
-            "\\RD_EN": _ast.Cat(port.en for port in fragment.read_ports),
-            "\\RD_ARST": _ast.Const(0, len(fragment.read_ports)),
-            "\\RD_SRST": _ast.Const(0, len(fragment.read_ports)),
-            "\\RD_ADDR": _ast.Cat(port.addr for port in fragment.read_ports),
-            "\\RD_DATA": _ast.Cat(port.data for port in fragment.read_ports),
+            "\\RD_EN": _ast.Cat(port._en for port in fragment._read_ports),
+            "\\RD_ARST": _ast.Const(0, len(fragment._read_ports)),
+            "\\RD_SRST": _ast.Const(0, len(fragment._read_ports)),
+            "\\RD_ADDR": _ast.Cat(port._addr for port in fragment._read_ports),
+            "\\RD_DATA": _ast.Cat(port._data for port in fragment._read_ports),
             "\\WR_CLK": _ast.Cat(wr_clk),
-            "\\WR_EN": _ast.Cat(_ast.Cat(en_bit.replicate(port.granularity) for en_bit in port.en) for port in fragment.write_ports),
-            "\\WR_ADDR": _ast.Cat(port.addr for port in fragment.write_ports),
-            "\\WR_DATA": _ast.Cat(port.data for port in fragment.write_ports),
+            "\\WR_EN": _ast.Cat(make_en(port) for port in fragment._write_ports),
+            "\\WR_ADDR": _ast.Cat(port._addr for port in fragment._write_ports),
+            "\\WR_DATA": _ast.Cat(port._data for port in fragment._write_ports),
```

Comment:
I thought I removed leading `\\`--is there another bug somewhere?

Reply:
didn't you do this on the new-ir branch only?

Reply:
oh oops